### ISSUE TITLE
Label function sometimes calling sometimes not, intermittent issue

### ISSUE THREAD

**Issue by peeyushsrivastava** [recharts/recharts#1135]
I am using Recharts on my dashboard, whenever user login first on the dashboard , it shows no label, no animation but when we refresh the page it started working. Even after implementing all comments with #829 , it didn't work. I think there is some issue with the library where it got stuck when we pass some dynamic props to the charts. I am trying to dig out in the library in between.

**Comment by shreniksv** [comment]
I am also facing the same issue. Labels do not render on initial render. It works fine when animation is disabled but I would like to have both animation and labels.

**Comment by jks8787** [comment]
I am also having this issue sadly

**Comment by hc1skalyanasundaram** [comment]
Are there any updates to this? I'm also having this issue

**Comment by sloev** [comment]
we have the issue as well just not in initial load.
in our case the chart renders but not the labellist on the bars.

**Comment by bavicj** [comment]
We have same issue as @sloev . On first load (page refresh), label render function is called and everything is displayed. Then changing to some other route and going back, data are fetched, but label render function is not called and no labels are displayed. Probably the same issue - #806

**Comment by carlobernardini** [comment]
It seems the issue is still there (using 1.2.0), the function is not called on initial render when the animation is done.

**Comment by hedshafran** [comment]
Experiencing this too. version **1.3.4**.

**Comment by Romasato** [comment]
It is an issue still in 1.3.5. 
I've basically do not need animations on my bars, so I've set on each <Bar isAnimationActive={false} />. That seems to do the trick.
Otherwise, if animation is still required, then I am not sure what's the workaround.

**Comment by hedshafran** [comment]
I use this workaround:

```
constructor() {
    super();
    this.state = { animate: true };
}

getSnapshotBeforeUpdate(prevProps) {
    const shouldAnimate = !equals(prevProps.data, this.props.data); // I used Ramda equals function - https://ramdajs.com/docs/#equals
    return shouldAnimate;
}

componentDidUpdate(prevProps, prevState, shouldAnimate) {
    if (prevState.animate !== shouldAnimate) {
      this.setState({ animate: shouldAnimate });
    }
}

// And then pass the updated state in the render function:
...
<Bar ... isAnimationActive={this.state.animate}>
...
```

This way, if the data hasn't changed, I'm passing `false` to the `isAnimationActive` prop.

**Comment by samotoo** [comment]
Well, still having the issue in 1.4.2, no label is showing (the valueAccessor function is not called at all) on initial render (page refresh), as long as I resize the window, the label is showing correctly.
Sadly I have to disable the animation on bar as a workaround. Hope this issue gets resolved.

**Comment by dmitryplatonov** [comment]
I think this issue is first raised there: https://github.com/recharts/recharts/issues/829
Too bad nobody is working on this issue.

**Comment by loopbender** [comment]
Having this issue in 1.5.0. 

Disabling animation solves the problem, however I do need to make use of animations and couldn't make @hedshafran's solution work to enable them.

**Comment by fabiosantoscode** [comment]
Can reproduce in 1.6.2

Additionally the label disappears sometimes when resizing (since we're using a ResponsiveContainer the component updates on resize)

**Comment by sanehab** [comment]
Yea, i'm using 1.6.2 and still facing the same issue, here is how i hacked it

`
class GraphWrapper extends React.Component {
    constructor() {
        this.state = {
            initialRender: true,
            graphData = []
        }
    }

    static getDerivedStateFromProps(props, state) {
        if (state.initialRender) {
          return {
            graphData: props.data, // the props that was meant to be the data given to the graph  
            initialRender: false,
          };
        }
    
        return null;
      }

    render() {
        // this works now
        return <BarChart data={this.state.graphData}/>
    }
}
`

**Comment by HuangStanley050** [comment]
Ran into the same issue, thanks @hedshafran for your workaround.

**Comment by igodorogea** [comment]
recharts@1.8.2 still not working.

**Comment by phara0hcom** [comment]
I have a better solution then @hedshafran, it started to work when I used [React.memo](https://reactjs.org/docs/react-api.html#reactmemo) function to check if the data and other props I send are not the same so it will not re-render if the data is the same. If you are using a React stateful component then you can use [shouldComponentUpdate](https://reactjs.org/docs/react-component.html#shouldcomponentupdate)

**Comment by samfromaway** [comment]
I have the same issue:  isAnimationActive={false}   fixed it for me too.

**Comment by StevenAccardo** [comment]
Recharts 1.8.5 still has this issue. This shouldn't have been closed.

**Comment by Othmanosx** [comment]
Recharts v2.0.10 the issue is still present, I had to disable the animations for it to work.
This should really be fixed.

**Comment by JonneBR** [comment]
Only works when isAnimationActive={false}. Should open this issue again because its still present.

**Comment by williamisnotdefined** [comment]
Yes, I'm having this same issue in the last day of 2021 hehe.
Setting isAnimationActive to false the problem is solved.

**Comment by maykl4iv** [comment]
This is amazing but I still having this issue, version is 2.4.1. Works with `isAnimationActive={false}`. Just curious if it will be fixed ever, it is like 5 years from when issue opened and it isn't fixed, and there is no update why

**Comment by ckifer** [comment]
It stems from this issue which remains open https://github.com/recharts/react-smooth/issues/44

Yes it can be fixed but there is no one working on recharts full-time, we have a few consistent maintainers who are contributing when they can. If anyone wants to track the bug down and fix it in react-smooth then I'll be happy to update it here. I can reopen this

**Comment by ckifer** [comment]
Change that _should_ resolve this is released in 2.9.0 but I cannot reproduce so I cannot test - if someone knows how to make this happen can you test it with the latest release?

**Comment by ckifer** [comment]
Finally was able to reproduce https://codesandbox.io/s/pie-chart-with-customized-label-forked-kqnrzg?file=/src/App.tsx
But only in recharts < 2.1.13 which it seems is where this was patched

Update in 2.9 seems to make no difference, but issue is resolved in newer versions of recharts either way. Closing this out
